src/AspNetCoreModuleV2/InProcessRequestHandler/inprocessapplication.h

```diff
@@ -133,53 +145,38 @@ class IN_PROCESS_APPLICATION : public InProcessApplicationBase
 
     PFN_ASYNC_COMPLETION_HANDLER    m_AsyncCompletionHandler;
 
-    // The event that gets triggered when managed initialization is complete
-    HandleWrapper<InvalidHandleTraits> m_pInitalizeEvent;
-
-    STRU                            m_struExeLocation;
+    std::wstring                    m_dotnetExeKnownLocation;
 
-    // The exit code of the .NET Core process
-    INT                             m_ProcessExitCode;
+    std::atomic_bool                m_blockManagedCallbacks;
+    bool                            m_Initialized;
+    bool                            m_waitForShutdown;
```

Comment:
volatile>

Reply:
Why?

Reply:
I think there is a race condition if it isn't. LoadManagedApplication sets this to false and ExecuteApplication reads the wrong value and tries to wait for shutdown.

Reply:
Or std::atomic_bool

Reply:
But there is SetEvent/WaitForSingleObject in between that takes care of memory synchronization. And if clr thread exits before the event is set we don't care.